sum_test.go

```diff
@@ -177,3 +180,46 @@ func TestBasicSum(t *testing.T) {
 		}
 	}
 }
+
+var sink []byte
```

Comment:
I think this has to be an exported type, if you want to disable certain compiler optimizations.

Reply:
Your point make sense, but I don't think the compiler is that smart yet.

Reply:
I think it is. At least I've read that advice in some benchmarking tutorial (can't find the link though).

Reply:
Fixed